coderd/authorize.go

```diff
@@ -27,6 +27,11 @@ func AuthorizeFilter[O rbac.Objecter](api *API, r *http.Request, action rbac.Act
 	return objects, nil
 }
 
+type HTTPAuthorizer struct {
+	Authorizer rbac.Authorizer
+	Logger     slog.Logger
```

Comment:
It's unclear why a logger is necessary here. Seems like passing an `rbac.Authorizer` via `Options` would suffice.

Reply:
used on line 67 to log authorization errors